_posts/2018-01-19-dockerize a symfony application.markdown

```diff
@@ -0,0 +1,107 @@
+---
+layout: post
+title:  "Dockerize a Symfony dev environment"
+date:   2018-01-19 00:00:00
+categories: symfony docker
+---
+
+We are going to run the [Symfony demo application](https://github.com/symfony/demo) in a docker container. The main purpose of this guide is to show how you can do this step by step. Afterwards, you must be able to dockerize any PHP application, or actually any application.
+
+If you haven't done it, start by installing [Docker](https://www.docker.com/community-edition). Even if you are not familiar with Docker, this guide is a good starting point to see how it works.
+
+Start by pulling the PHP image, running a container, and executing `bash` inside the PHP container. We can do this using a single command.
+
+````bash
+docker run -it php:fpm bash
+````
+You should get an output similar to
+````bash
+Unable to find image 'php:fpm' locally
+fpm: Pulling from library/php
+e7bb522d92ff: Pull complete
+...
+26b263fd4e72: Pull complete
+Digest: sha256:6907a969eed2e673584f5dac8189c8039f35d67ff273ebf99fa539abc32354c0
+Status: Downloaded newer image for php:fpm
+root@e5b799b92f6e:/var/www/html#
+````
+To confirm that you have a working PHP environment
+````bash
+root@e5b799b92f6e:/var/www/html# php -v
+PHP 7.2.1 (cli) (built: Jan  8 2018 23:39:24) ( NTS )
+Copyright (c) 1997-2017 The PHP Group
+Zend Engine v3.2.0, Copyright (c) 1998-2017 Zend Technologies
+````
+
+Now we head to [https://github.com/symfony/demo](https://github.com/symfony/demo) to check the installation which is just a single command
+````bash
+composer create-project symfony/symfony-demo
+````
+
+If you try to execute it in the container, you will get an error `bash: composer: command not found`.
+And this is exactly the main workflow to dockerize an application. Most of the time you are looking for `XXX not found` like error messages, and fix them. You will not install anything in the created PHP container because all changes you may make there will be lost when you restart the container. Instead you should create a *Dockerfile* so you can recreate (build) and run a container with all the changes already applied.
+`exit` the running container. Create a *symfony-demo* directory and create a Dockerfile inside it.
+
+Go to https://getcomposer.org/doc/faqs/how-to-install-composer-programmatically.md and copy the content of the script found there into *install-composer.sh*.
```

Comment:
Composer install can be done with a single line in dockerfile:
`RUN curl https://getcomposer.org/installer | php -- --filename=composer --install-dir=/bin`